google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPoolOptions.java

```diff
@@ -105,11 +104,12 @@ private SessionPoolOptions(Builder builder) {
     this.randomizePositionQPSThreshold = builder.randomizePositionQPSThreshold;
     this.inactiveTransactionRemovalOptions = builder.inactiveTransactionRemovalOptions;
     this.poolMaintainerClock = builder.poolMaintainerClock;
-    // TODO: Remove when multiplexed sessions are guaranteed to be supported.
+    // useMultiplexedSession priority => Environment var > private setter > client default
```

Comment:
We should change this priority to `private setter -> Env var -> client default`. Otherwise, it will be impossible to use this feature in for example JDBC and PGAdapter, as it is not possible to set an environment variable in Java after the JVM has started. So instead, JDBC and PGAdapter will be using the private setter to override the default. This is also the reason that we still need the escape hatch in the form of an environment variable that __disables__ multiplexed sessions.

Reply:
So the behavior should be:
1. If `GOOGLE_CLOUD_SPANNER_FORCE_DISABLE_MULTIPLEXED_SESSIONS=true` => Use regular sessions.
2. If the private setter has been called => Use the value from the private setter.
3. If `GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS` has been set => Use the value from that env var.
4. Use the default in the client library (currently `false`)

Reply:
@olavloite 
1. The value in environment variable overrides the private setter only when the env `GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS` is **set**. In case of JDBC and PGAdapter, the env will not be set and the private setter will override the default behavior.
2. For escape hatch they can set `GOOGLE_CLOUD_SPANNER_MULTIPLEXED_SESSIONS` to `false` which will disable multiplexed sessions. I don't understand why we need to maintain a seperate environment variable `GOOGLE_CLOUD_SPANNER_FORCE_DISABLE_MULTIPLEXED_SESSIONS` for disabling?

Reply:
Yeah, I think you are right. So the working will be:
1. PGAdapter / JDBC use the private setter to enable mux sessions by default.
2. As long as no env var has been set, the private setter value is respected.
3. Users can disable mux sessions by setting the env var to `false`.

Setting the env var to `false` will also work for the client library itself in the future, even after the default in the client lib has become true.